src/Symfony/Component/Webhook/Client/RequestParserInterface.php

```diff
@@ -24,11 +24,11 @@ interface RequestParserInterface
     /**
      * Parses an HTTP Request and converts it into a RemoteEvent.
      *
-     * @return ?RemoteEvent Returns null if the webhook must be ignored
+     * @return RemoteEvent|RemoteEvent[]|null Returns null if the webhook must be ignored
      *
      * @throws RejectWebhookException When the payload is rejected (signature issue, parse issue, ...)
      */
-    public function parse(Request $request, #[\SensitiveParameter] string $secret): ?RemoteEvent;
+    public function parse(Request $request, #[\SensitiveParameter] string $secret): RemoteEvent|array|null;
```

Comment:
We could stay BC with another parameter enabling. This new behavior, couldn't we?

Reply:
Can you elaborate a bit on this?

Reply:
I was thinking about removing the return type and move it to `@return`, then in the method check, if an additional argument was provided like `$returnArrayInsteadOfOneObject` (ofc to verbose) and in this case return `RemoteEvent[] plus raising a deprecation

Reply:
The BC break is the new return type. I don't see how a parameter would help.

Reply:
```diff
+ /*
+  * @return RemoteEvent|RemoveEvent[]|null
+  */
- public function parse(Request $request, #[\SensitiveParameter] string $secret): RemoteEvent|array|null;
+ public function parse(Request $request, #[\SensitiveParameter] string $secret);
```

Even if we remove the return type?

Reply:
Still a BC break for consumers of the interface:

Consider the existing code in someone's app:

```php
if (!$event = $parser->parse($request, $secret)) {
    return ...;
}

$this->bus->dispatch(new ConsumeRemoteEventMessage($type, $event)); // error if $event is an array - this is the BC break
```